src/app/home/home.page.ts

```diff
@@ -47,43 +47,56 @@ export class HomePage {
   constructor(private countryService: CountryService) { }
 
   ngAfterViewInit(): void {
-    this.prepareSearchDenounce();
-    this.loadCountries();
-    this.loadFavourites();
-  }
-  private prepareSearchDenounce(): void {
-    this.searchBar.ionInput
-      .pipe(
-        debounceTime(300)
-      )
-      .subscribe(event => {
-        this.filterCountries((event.target as HTMLInputElement).value);
-      });
-  }
-
-  private loadCountries(): void {
-    this.countryService.getCountries().pipe(
-      catchError(error => this.handleError(error)),
-      tap(result => {
-        if (result && !result.error) {
-          this.countries = result.data;
-          this.baseCountries = result.data;
-          this.groupCountries();
-          this.createLetters();
-          this.baseBackupGroupedCountries = { ...this.groupedCountries };
-        }
-      })
-    ).subscribe();
+    Promise.all([this.prepareSearchDenounce(), this.loadCountries(), this.loadFavourites()]).catch((error) => {
+      this.handleError(error);
+    })
+  }
+  private prepareSearchDenounce(): Promise<void> {
+    return new Promise((resolve) => {
+      this.searchBar.ionInput
+        .pipe(
+          debounceTime(300)
+        )
+        .subscribe(event => {
+          this.filterCountries((event.target as HTMLInputElement).value);
+        });
+      resolve();
+    });
+  }
+
+  private loadCountries(): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this.countryService.getCountries().pipe(
+        catchError(async (error) => {
+          this.handleError(error);
+          reject(error);
+        }),
+        tap(result => {
+          if (result && !result.error) {
+            this.countries = result.data;
+            this.baseCountries = result.data;
+            this.groupCountries();
+            this.createLetters();
+            this.baseBackupGroupedCountries = { ...this.groupedCountries };
+            resolve();
+          }
+        })
+      ).subscribe();
+    });
+
   }
 
   private handleError(error: any): Observable<any> {
     console.error('Error loading countries:', error);
     return of(null);
   }
 
-  private loadFavourites() {
-    const storedFavourites = localStorage.getItem('favourites');
-    this.favourites = JSON.parse(storedFavourites ?? '[]');
+  private loadFavourites(): Promise<void> {
```

Comment:
creates a resolved promise immediately as below for simplify : 
private loadFavourites(): Promise<void> {
  return Promise.resolve().then(() => {
    const storedFavourites = localStorage.getItem('favourites');
    this.favourites = JSON.parse(storedFavourites ?? '[]');
  });
}